xlive/Blam/Engine/saved_games/cartographer_player_profile/cartographer_player_profile_weapon_offsets.cpp

```diff
@@ -0,0 +1,82 @@
+#include "stdafx.h"
+#include "cartographer_player_profile_weapon_offsets.h"
+
+#include "cache/cache_files.h"
+
+/* constants */
+
+const s_weapon_custom_offset_constant_data k_weapon_offset_constant_data[k_weapon_offset_weapon_count]
+{
+	{{{0.02f, 0.f, -0.004f}},		"objects\\weapons\\rifle\\battle_rifle\\battle_rifle"},
+	{{{0.02f, 0.f, 0.f}},			"objects\\weapons\\rifle\\beam_rifle\\beam_rifle"},
+	{{{0.02f, 0.f, 0.f}},			"objects\\weapons\\rifle\\brute_plasma_rifle\\brute_plasma_rifle"},
+	{{{0.02f, -0.01f, 0.001f}},		"objects\\weapons\\support_low\\brute_shot\\brute_shot"},
+	{{{0.02f, 0.f, -0.004f}},		"objects\\weapons\\rifle\\covenant_carbine\\covenant_carbine"},
+	{{{0.05f, 0.f, 0.f}},			"objects\\weapons\\melee\\energy_blade\\energy_blade"},
+	{{{0.02f, 0.f, -0.005f}},		"objects\\weapons\\support_high\\flak_cannon\\flak_cannon"},
+	{{{0.03f, 0.f, -0.005f}},		"objects\\weapons\\pistol\\magnum\\magnum"},
+	{{{0.03f, 0.f, -0.005f}},		"objects\\weapons\\pistol\\needler\\needler"},
+	{{{0.03f, 0.f, -0.005f}},		"objects\\weapons\\pistol\\plasma_pistol\\plasma_pistol"},
+	{{{0.02f, 0.f, 0.f}},			"objects\\weapons\\rifle\\plasma_rifle\\plasma_rifle"},
+	{{{0.02f, 0.f, -0.005f}},		"objects\\weapons\\support_high\\rocket_launcher\\rocket_launcher"},
+	{{{0.02f, 0.03f, 0.f}},			"objects\\characters\\sentinel_aggressor\\weapons\\beam\\sentinel_aggressor_beam"},
+	{{{0.02f, -0.001f, -0.015f}},	"objects\\weapons\\rifle\\shotgun\\shotgun"},
+	{{{0.02f, 0.f, 0.f}},			"objects\\weapons\\rifle\\smg\\smg"},
+	{{{0.01f, 0.f, 0.f}},			"objects\\weapons\\rifle\\sniper_rifle\\sniper_rifle"}
+};
+
+/* globals */
+
+static datum g_weapon_offset_tag_indices[k_weapon_offset_weapon_count] = 
+{
+	{ NONE },
```

Comment:
?

Reply:
why

Reply:
Actually, we need to do this since `= { NONE }` only initializes the 0th index